From 57ebcd6fc73ad90b7a3dce680ee9811a27ddcf8a Mon Sep 17 00:00:00 2001
From: Sameer Rahmani <lxsameer@gnu.org>
Date: Fri, 15 Sep 2017 14:02:03 +0100
Subject: [PATCH] CLJS-2166: Add uri? predicate

Clojure introduced a `uri?` predicate in 1.9. This commit adds support for `uri`
to Clojurescript under `cljs.core/uri?`. It returns true if the given argument
is an instance of `goog.Uri`, false otherwise.
---
 src/main/cljs/cljs/core.cljs      |  9 ++++++++-
 src/test/cljs/cljs/core_test.cljs | 13 ++++++++++++-
 2 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/src/main/cljs/cljs/core.cljs b/src/main/cljs/cljs/core.cljs
index 519e4ab0..bf5631b5 100644
--- a/src/main/cljs/cljs/core.cljs
+++ b/src/main/cljs/cljs/core.cljs
@@ -12,7 +12,8 @@
             [goog.string :as gstring]
             [goog.object :as gobject]
             [goog.array :as garray])
-  (:import [goog.string StringBuffer]))
+  (:import [goog.string StringBuffer]
+           goog.Uri))
 
 ;; next line is auto-generated by the build-script - Do not edit!
 (def *clojurescript-version*)
@@ -11168,3 +11169,9 @@ reduces them without incurring seq initialization"
   Bootstrap only."
   [ns-obj]
   (.-name ns-obj))
+
+(defn uri?
+  "Returns true X is a goog.Uri instance."
+  {:added "1.9"}
+  [x]
+  (instance? goog.Uri x))
diff --git a/src/test/cljs/cljs/core_test.cljs b/src/test/cljs/cljs/core_test.cljs
index f2d7b065..05b7aecd 100644
--- a/src/test/cljs/cljs/core_test.cljs
+++ b/src/test/cljs/cljs/core_test.cljs
@@ -15,7 +15,8 @@
             [clojure.test.check.properties :as prop :include-macros true]
             [clojure.string :as s]
             [clojure.set :as set]
-            [goog.object :as gobject]))
+            [goog.object :as gobject])
+  (:import goog.Uri))
 
 (deftest test-metadata
   (testing "Testing metadata"
@@ -1493,3 +1494,13 @@
   ;;   (assert (= (:arglists var-meta) '([a b]))))
 
   )
+
+(deftest uri-predicate
+  (testing "Testing uri?"
+    (is (not (uri? "http://clojurescript.org")))
+    (is (not (uri? 42)))
+    (is (not (uri? [])))
+    (is (not (uri? {})))
+    (is (uri? (goog.Uri. "")))
+    (is (uri? (goog.Uri. "http://clojurescript.org")))
+    (is (uri? (goog.Uri. "some string")))))
-- 
2.11.0

